frontend_vue/src/components/base/BaseTextField.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    class="flex flex-col text-grey-800 textfield-wrapper"
+    :class="{ 'w-full': fullWidth }"
+  >
+    <label
+      :for="id"
+      class="mb-4 ml-4 font-semibold"
+      >{{ label }}
+      <span
+        v-if="required"
+        class="text-green-500"
+        >*</span
+      ></label
+    >
+    <component
```

Comment:
what is a 'component'?

Reply:
it's a dynamic Component. You can switch between different vue components/html components via the attribute value `:is`
In this case
`<component :is="input | textarea" ...`

Textarea and input share all their properties, so in this case I just told vue to display one or the other depending on  the 'multiline' prop ( is multiline? the component should be a textarea. Is not multiline? It's just an input )